modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/LiferayEditor.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+class LiferayEditor {
+	constructor(
+		contents,
+
+		// editor,
+
+		initMethod,
+		name,
+
+		// onBlurMethod,
+		onChangeMethod,
+		// onFocusMethod,
+		// onInitMethod
+
+	) {
+		this.contents = contents;
+
+		// this.editor = editor;
+
+		this.initMethod = initMethod;
+		this.name = name;
+
+		// this.onBlurMethod = onBlurMethod;
+
+		this.onChangeMethod = onChangeMethod;
+
+		// this.onFocusMethod = onFocusMethod;
+		// this.onInitMethod = onInitMethod;
+
+		this.instanceReady = false;
+
+		this.ckEditor = CKEDITOR.instances[this.name];
```

Comment:
Sometimes CKEDITOR global is not defined. For example, when using pure `ckeditor.jsp` file, without using a React component for BalloonEditor for e.g, it works. However, when loading inside ballooneditor JSP, It fails :/